lib/Sema/CSSimplify.cpp

```
@@ -9211,13 +9206,16 @@ bool ConstraintSystem::recordFix(ConstraintFix *fix, unsigned impact) {
 
   if (isAugmentingFix(fix)) {
     Fixes.push_back(fix);
-  } else {
+  } else if (auto *anchor = getAsExpr(fix->getAnchor())) {
```

Should we assert that a non-augmenting fix has an expr anchor? Otherwise we'll silently drop it here

---

Actually I think I'll just refactor this code to avoid depending on anchor being an expression.

---

Done, @hamishknight how does that look?

---

Looks great, thanks!